apps/assessments/models.py

```diff
@@ -0,0 +1,170 @@
+# Django Imports
+from django.conf import settings
+from django.db import models
+
+# HTK Imports
+from htk.utils import htk_setting
+
+
+class AbstractAssessment(models.Model):
+    """Main class for Assessments app.
+
+    Possible use cases include:
+    - Quizzes and Exams
+    - Surveys
+    """
+
+    name = models.CharField(max_length=255, blank=True)
+    # Tracks the version of the assessment
+    version = models.IntegerField(default=1)
+    # Number of attempts allowed: 0 for unlimited attempts, 1 for only 1.
+    num_allowed_attempts = models.PositiveIntegerField(default=0)
+    # when `True` and `num_allowed_attempts` == 1, previous responses are overwritten
+    is_repeat_allowed = models.BooleanField(default=False)
+    # Optional "Exit message" to display when an incorrect answer is provided
+    knockout_message = models.TextField(blank=True, null=True)
+
+    class Meta:
+        abstract = True
+
+    def __str__(self):
+        return f"{self.name} (Version {self.version})"
+
+
+class AbstractAssessmentQuestion(models.Model):
+    """Represents one question in an Assessment."""
+
+    ASSESSMENT_MODEL = htk_setting('HTK_ASSESSMENT_MODEL')
+
+    QUESTION_TYPES = (
+        ('FR', 'Free Response'),
+        ('MC', 'Multiple Choice'),
+        ('YN', 'Yes or No'),
+    )
+
+    assessment = models.ForeignKey(
+        ASSESSMENT_MODEL, related_name='questions', on_delete=models.CASCADE
+    )
+    text = models.TextField()
```

Comment:
Also Let's add a `max_length` to this one please? I am thinking of something like 500 or 750?

Here is a sample 1000 characters text:


In the heart of the bustling city, nestled among towering skyscrapers, lies a tranquil park. This oasis, surrounded by the constant hum of urban life, serves as a refuge for both city dwellers and wildlife. On a crisp autumn day, the park is awash with the hues of amber and gold, leaves rustling softly underfoot as people meander along its winding paths. Benches, sporadically placed, offer rest and a moment to immerse in the serenity. Here, an elderly man feeds pigeons from a crumpled paper bag, his eyes twinkling with a quiet joy. Nearby, a young mother watches her child take tentative steps on the grass, her laughter mingling with the breeze. As the sun begins to dip below the skyline, casting long shadows and bathing everything in a golden glow, the park transforms into a canvas of light and shadow. This small patch of nature, amidst the steel and concrete, remains a testament to the city's complex character—a blend of frenzy and calm, where every corner offers a story waiting to be discovered.

Reply:
Set `max_length` as `256` since we limit to only `120` chars.